lib/main.coffee

```diff
@@ -5,22 +5,33 @@ module.exports =
       default: ''
       type: 'string'
       description: 'Leave empty to use bundled'
+    lintInlineJavaScript:
+      type: 'boolean'
+      default: false
+      description: 'Lint JavaScript inside `<script>` blocks in HTML or PHP files'
+
   provideLinter: ->
     if process.platform is 'win32'
       jshintPath = require('path').join(__dirname, '..', 'node_modules', '.bin', 'jshint.cmd')
     else
       jshintPath = require('path').join(__dirname, '..', 'node_modules', '.bin', 'jshint')
+    scopes = ['source.js', 'source.js.jsx']
+    lintInlineJavaScript = atom.config.get('linter-jshint.lintInlineJavaScript')
+    if lintInlineJavaScript is true
+      scopes.push('source.js.embedded.html')
     helpers = require('atom-linter')
     reporter = require('jshint-json') # a string path
     provider =
-      grammarScopes: ['source.js', 'source.js.jsx']
+      grammarScopes: scopes
       scope: 'file'
       lintOnFly: true
       lint: (textEditor) ->
         exePath = atom.config.get('linter-jshint.jshintExecutablePath') || jshintPath
         filePath = textEditor.getPath()
         text = textEditor.getText()
-        parameters = ['--reporter', reporter, '--extract', 'auto', '--filename', filePath, '-']
+        parameters = ['--reporter', reporter, '--filename', filePath, '-']
+        if lintInlineJavaScript is true and ~textEditor.getGrammar().scopeName.indexOf 'text.html'
+          parameters.splice(2, 0, '--extract', 'always')
```

Comment:
I know it's needed to let it work correctly, because it means that jshint should expect input from stdin instead of a file or something, but I was saying that you don't add it to the parameters array on declaration and you push it last on the line below, how does that sound?